pages/agent/plugins.md.erb

```diff
@@ -0,0 +1,124 @@
+
+# Buildkite Agent Plugins
+
+__Plugins are only available in the beta 3.0 version of the agent__
+
+Plugins provide a way to extend the functionality that Buildkite natively offers. Essentially, they are neatly packaged collections of [Hooks](hooks) that run as part of your builds. Buildkite offers a [collection of useful plugins](https://github.com/buildkite-plugins) for things like Docker integration, Github Releases and interaction with Amazon Web Services and Google Cloud. 
+
+<%= toc %>
+
+## Using a plugin
+
+Plugins are only available for [pipelines] expressed in yaml, so all of these examples are going to be using that syntax.  
+
+Plugins are defined per-step and are automatically fetched from github based on the name of the project. For instance, checkout our clown-detection plugin which scans your code for pesky clowns:
+
+```yml
+steps:
+  - plugins:
+      detect-clowns:
+        emojis: 🤡
+```
+
+This example will install [buildkite-plugins/detect-clowns-buildkite-plugin](https://github.com/buildkite-plugins/detect-clowns-buildkite-plugin) and run it for this build step.
+
+You could just as easily write your own plugin on GitHub at `myusername/blah-buildkite-plugin` and it would be available as `myusername/blah` in your pipeline.yml.
+
+Another example of something that we use every day is the [docker-compose](https://github.com/buildkite-plugins/docker-compose-buildkite-plugin) plugin. It makes it easy to run docker-compose in your build steps:
+
+```yml
+steps:
+  # Prebuild the app image, upload it to a registry for later steps
+  - name: "Docker Build"
+    plugins:
+      docker-compose#v1.3.2:
+        build: app
+        image-repository: index.docker.io/org/repo
+
+  - wait
+
+  # Use the app image built above to run concurrent tests
+  - name: "Docker Test %n"
+    command: test.sh
+    parallelism: 25
+    plugins:
+      docker-compose#v1.3.2:
+        run: app
+```
+
+In the 2.x agent, this sort of thing was only possible by either using the limited docker environment variables, or writing a custom script. 
+
+## Anatomy of a Plugin
```

Comment:
This section seems to be information for people who are writing plugins? Anything globally applicable to using plugins or working with them could be pulled out into the 'using' section or the intro. All the rest could be moved down into the 'writing' section?